### DISCUSSION TITLE
How to retrieve data from the server after client-side fields have been updated

### DISCUSSION THREAD

**Discussion by smo043** [vercel/next.js#44709] [Help] [answered]
Hello, we are using the most recent version of NextJS with the app directory. In one scenario, we are using server components to retrieve data from the initial payload. 

However, in another scenario, we have a few fields that can be updated on the client side; as soon as these fields are updated, we must call the same api that we did in the initial payload. How can the API call be used in this case? 

Is it still possible to use the server components API? Or will it be a client-side call?

Please advise. Thanks

**Comment by icyJoseph** [comment, marked as answer]
Mmm I think you can refactor your code a bit so that the API call you make within your server component is parametrized and can be reused elsewhere - why? Well because with the current state of things, I think you need to create a `pages/api` file, with a function within it, this would be an endpoint that your client calls to do CRUD, and it'd be just easier if its the same function being called in the API file and in the Server Components.

With parametrized I mean that the request uses `fetch`, and that it is wrapped in a function that takes as inputs values we can calculate from within Server Components or the API route.

**Comment by smo043** [comment]
@icyJoseph - Thank you, we will follow the same approach you described. Appreciate your help.

**Reply by smo043** [reply]
And not to forget 'router.refresh()' does the job to refresh the data from the server without page load.
